### PR TITLE
Julian’s attempt at logging modifications

### PR DESCRIPTION
**Pull request by julianlai777**

### CODE REVIEW

File: src/common/command-factory.cpp [hunk 1]
```diff
@@ -10,9 +10,13 @@
 ICommand* CommandFactory::CreateCommand(char *data) {
     ICommand* result = 0;
 
-    if (!data) { 
-        fprintf(stderr, "NULL argument passed to CreateCommand() in %s\n", __FILE__); // TODO log 
-        return NULL; 
+    if (!data) {
+        
+        char buffer[CS1_MAX_LOG_ENTRY];
```
Comment (SpaceShawn) [inline src/common/command-factory.cpp]: Declare outside of conditional

File: src/common/command-factory.cpp [hunk 2]
```diff
@@ -10,9 +10,13 @@
...
+        memset(buffer, 0, CS1_MAX_LOG_ENTRY);
+        snprintf(buffer, "NULL arguemnt passed to CreateCommand() in %s/n", __FILE__);
```
Comment (SpaceShawn) [inline src/common/command-factory.cpp]: Clear out all line breaks (/n)

File: src/common/decode-command.cpp [hunk 3]
```diff
@@ -20,11 +20,20 @@ void* DecodeCommand::Execute() {
...
     }
-
-    printf("  GetSrcPath:   %s\n", this->GetSrcPath());
-    printf("  GetDestPath:  %s\n", this->GetDestPath());
-    printf("  IsExecutable: %d\n", this->IsExecutable());
-    fflush(stdout);
+    
+    char std_buffer[CS1_MAX_LOG_FILE_SIZE];
+    memset(std_buffer, 0, CS1_MAX_LOG_FILE_SIZE);
+    snprintf(std_buffer, CS1_MAX_LOG_FILE_SIZE, "  GetSrcPath:   %s\n", this->GetSrcPath());
```
Comment (SpaceShawn) [inline src/common/decode-command.cpp]: Same thing with line breaks

File: src/common/decode-command.cpp [hunk 4]
```diff
@@ -91,14 +103,16 @@ InfoBytes* DecodeCommand::ParseResult(char *result)
 
     char buffer[100];
```
Comment (SpaceShawn) [inline src/common/decode-command.cpp]: Define a macro that is descriptive instead of magic 100

File: src/common/decode-command.cpp [hunk 5]
```diff
@@ -91,14 +103,16 @@ InfoBytes* DecodeCommand::ParseResult(char *result)
...
     if(info_bytes.decode_status == CS1_SUCCESS)
-    {    
-        snprintf(buffer,100,"Decode success: No informaation to report");
-        Shakespeare::log(Shakespeare::NOTICE,cs1_systems[CS1_COMMANDER], buffer);
+    {
+        memset(buffer, 0, 100);
```
Comment (SpaceShawn) [inline src/common/decode-command.cpp]: No magic numbers

File: src/common/command-factory.cpp (hunk 1 above)
Comment (CandyRedTele) [inline src/common/command-factory.cpp]: rename to log_buf

File: src/common/decode-command.cpp [hunk 6]
```diff
@@ -20,11 +20,20 @@ void* DecodeCommand::Execute() {
         fpSrcFile = fopen(this->GetSrcPath(), "rb");
         retry =- 1;
     }
-
-    printf("  GetSrcPath:   %s\n", this->GetSrcPath());
-    printf("  GetDestPath:  %s\n", this->GetDestPath());
-    printf("  IsExecutable: %d\n", this->IsExecutable());
-    fflush(stdout);
+    
+    char std_buffer[CS1_MAX_LOG_FILE_SIZE];
```
Comment (CandyRedTele) [inline src/common/decode-command.cpp]: rename to log_buf

**Comment by SpaceShawn** [comment]
How is this coming?

**Pull request closed without merging**